pkg/scalers/redis_streams_scaler.go

```diff
@@ -166,6 +172,42 @@ func createEntriesCountFn(client redis.Cmdable, meta *redisStreamsMetadata) (ent
 			}
 			return entriesLength, nil
 		}
+	case lagFactor:
+		entriesCountFn = func(ctx context.Context) (int64, error) {
+			// Make sure that redis is version 7+, which is required for xinfo lag
+			info, err := client.Info(ctx).Result()
+			if err != nil {
+				err := errors.New("could not find Redis version")
+				return -1, err
+			}
+			infoLines := strings.Split(info, "\n")
+			versionLine := infoLines[1]
```

Comment:
Would it not be better to match `redis_version` prefix from the strings here to ensure that even if the version isn't present on line 1, the logic to extract the version works?